packages/core/src/connectors/facebook/index.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * Reference: Manually Build a Login Flow
+ * https://developers.facebook.com/docs/facebook-login/manually-build-a-login-flow
+ */
+import got, { RequestError as GotRequestError } from 'got';
+import { stringify } from 'query-string';
+import { z } from 'zod';
+
+import {
+  accessTokenEndpoint,
+  authorizationEndpoint,
+  scope,
+  userInfoEndpoint,
+} from '@/connectors/facebook/constant';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorMetadata,
+  ConnectorType,
+  GetAccessToken,
+  GetAuthorizationUri,
+  GetUserInfo,
+  ValidateConfig,
+} from '@/connectors/types';
+import { getConnectorConfig, getConnectorRequestTimeout } from '@/connectors/utilities';
+
+export const metadata: ConnectorMetadata = {
+  id: 'facebook',
+  type: ConnectorType.Social,
+  name: {
+    en: 'Sign In with Facebook',
+    zh_CN: 'Facebook 登录',
+  },
+  logo: './logo.png',
+  description: {
+    en: 'Sign In with Facebook',
+    zh_CN: 'Facebook 登录',
+  },
+};
+
+const facebookConfigGuard = z.object({
+  clientId: z.string(),
+  clientSecret: z.string(),
+});
+
+type FacebookConfig = z.infer<typeof facebookConfigGuard>;
+
+export const validateConfig: ValidateConfig = async (config: unknown) => {
+  if (!config) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, 'Missing config');
+  }
+
+  const result = facebookConfigGuard.safeParse(config);
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error.message);
+  }
+};
+
+export const getAuthorizationUri: GetAuthorizationUri = async (redirectUri, state) => {
+  const config = await getConnectorConfig<FacebookConfig>(metadata.id);
+
+  return `${authorizationEndpoint}?${stringify({
+    client_id: config.clientId,
+    redirect_uri: redirectUri,
+    response_type: 'code',
+    state,
+    scope, // Only support fixed scope for v1.
+  })}`;
+};
+
+export const getAccessToken: GetAccessToken = async (code, redirectUri) => {
+  type FacebookAccessTokenResponse = {
+    access_token: string;
+    token_type: string;
+    expires_in: number;
+  };
+
+  const { clientId: client_id, clientSecret: client_secret } =
+    await getConnectorConfig<FacebookConfig>(metadata.id);
+
+  const { access_token: accessToken } = await got
+    .post({
+      url: accessTokenEndpoint,
+      form: {
+        code,
+        client_id,
+        client_secret,
+        redirect_uri: redirectUri,
+      },
+      timeout: await getConnectorRequestTimeout(),
+    })
+    .json<FacebookAccessTokenResponse>();
+
+  if (!accessToken) {
+    throw new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid);
+  }
```

Comment:
nit: maybe use `assertThat` instead as Simeng mentioned?

Reply:
See [comment](https://github.com/logto-io/logto/pull/321#discussion_r820402976)

Reply:
Took a look at the comment you gave, I think they are not the same issue.
In this case, the code snippet can be written as:
`assertThat(access_token, new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid));`

Reply:
My bad. Updated. PTAL.

Related PR #333